src/api/controllers/WalletController.js

```diff
@@ -22,7 +24,14 @@ class WalletController extends PointSDKController {
     }
 
     publicKey() {
-        const publicKeyBuffer = ethereumjs.privateToPublic(this.defaultWallet.privateKey);
+        // '0x' should be removed
+        const privateKeyBuffer = Buffer.from(this.defaultWallet.privateKey.slice(2), 'hex');
+        const isValidPrivate = ethereumjs.isValidPrivate(privateKeyBuffer);
+        if (!isValidPrivate) {
+            throw Error('invalid private key');
+        }
+
+        const publicKeyBuffer = ethereumjs.privateToPublic(privateKeyBuffer);
```

Comment:
@nicobevilacqua Maybe its more suitable to move this conversion functionality to the `blockchain` module?

Reply:
Ok, I can move it. I wasn't sure were it would fit better.